File: 66-PlusOne/66-PlusOne.dart
// Last updated: 3/30/2026, 10:25:54 AM
1class Solution {
2  List<int> plusOne(List<int> digits) {
3    for (int i = digits.length - 1; i >= 0; i--) {
4      if (digits[i] == 9) {
5        digits[i] = 0;
6      } else {
7        digits[i]++;
8        return digits;
9      }
10    }
11
12    digits.insert(0, 1);
13    return digits;
14  }
15}

File: 66-PlusOne/66-PlusOne.py
# Last updated: 3/30/2026, 10:42:13 AM
1class Solution:
2    def plusOne(self, digits: List[int]) -> List[int]:
3
4     
5
6        for i in range(len(digits)-1, -1, -1):
7            if digits[i] + 1 != 10:
8                digits[i] +=1
9                return digits
10
11            digits[i] =0
12
13            if i == 0:
14                return [1]+digits
15
16
17
18
19
20        